spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  include Pundit
+
+  describe '#new' do
+    login_user
+
+    it 'renders template :new' do
+      get :new
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe '#create' do
+    def do_request
+      post :create, params: post_params
+    end
+
+    login_user
+
+    context 'valid params' do
+      let(:post_params) { { post: attributes_for(:post) } }
+
+      it 'creates successfully' do
+        expect { do_request }.to change { Post.count }.by(1)
+        expect(flash[:notice]).to eq('The post has been created')
+        expect(response).to redirect_to post_path(Post.last)
+      end
+    end
+
+    context 'invalid params' do
+      let(:post_params) { { post: attributes_for(:post, title: nil) } }
+
+      it 'creates unsuccessfully' do
+        expect { do_request }.to change { Post.count }.by(0)
+        expect(response).to render_template(:new)
+      end
+    end
+  end
+
+  describe '#show' do
+    login_user
+    let!(:usr_post) { create(:post) }
+
+    it 'show post from users' do
+      get :show, params: { id: usr_post }
+      expect(response).to render_template(:show)
+      expect(assigns(:comment)) == usr_post.comments.build
```

Comment:
change to `expect(assigns(:comment)).to eq(usr_post.comments.build)`.

Reply:
rspec fail if switch to use **eq** instead of **==**
![Uploading 

![image](https://user-images.githubusercontent.com/54341400/70203515-b719d480-174f-11ea-83c7-8cf04e71be47.png)